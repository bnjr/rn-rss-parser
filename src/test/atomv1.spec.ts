import * as rssParser from '../index';

// Import feed XML samples directly as CommonJS
const atomv1 = require('./samples/atomv1');
const atomv1WithItunes = require('./samples/atomv1-with-itunes');
const atomv1NoUpdated = require('./samples/atomv1-no-updated');
const huffpost = require('./samples/huffpost');

describe('when parse ATOM', () => {
  describe('valid document', () => {
    it('should return feed items', async () => {
      const result = rssParser.parseFeed(atomv1.feed);

      expect(result.title).toBe('ATOM title');
      expect(result.url).toBe('http://bakery-store.example.com/');
      expect(result.description).toBe('A sample ATOM feed');
      expect(result.items.length).toBe(2);
      expect(result.image?.url).toBe(
        'https://b.thumbs.redditmedia.com/ntr1FkBiO3nk4t4Vgy5GXoPQ_j2hirENH9iT8rXNf8M.png',
      );
      expect(result.items[0].title).toBe('Where Did The Cookie Come From');
      expect(result.items[0].id).toBe(
        'http://bakery-store.example.com/information/2016/01/02/where-did-the-cookie-come-from',
      );
      expect(result.items[0].image?.url).toBe(undefined);
      expect(result.items[0].url).toBe(
        'http://bakery-store.example.com/information/2016/01/02/where-did-the-cookie-come-from.html',
      );
      // AtomFeedItem does not expose rel directly, so skip rel check
      expect(result.items[0].media.length).toBe(1);
      expect(result.items[0].media[0].url).toBe('https://www.example.com/audio.mp3');
      expect(result.items[0].media[0].mimeType).toBe('audio/mpeg');
      expect(String(result.items[0].media[0].length)).toBe('1234');
      expect(result.items[0].description).toBe(
        'The chocolate chip cookie was invented by Ruth Graves Wakefield.',
      );
      expect(result.items[1].title).toBe('What Is Sour Dough');
      expect(
        result.items[0].published instanceof Date
          ? result.items[0].published.toISOString()
          : result.items[0].published,
      ).toContain('2016-01-01');
      expect(result).toMatchSnapshot();
    });
  });

  describe('with itunes elements', () => {
    it('should return itunes information for channel and item elements', async () => {
      const result = rssParser.parseFeed(atomv1WithItunes.feed);
      // The new model may not expose iTunes fields directly; skip or check extensions if implemented
      // expect(result.extensions?.itunes).not.toBe(undefined); // Uncomment if extensions are implemented
      expect(result.items.length).toBe(2);
      // You may add more checks here if your model exposes iTunes fields
      expect(result).toMatchSnapshot();
    });
  });

  describe('verifying a HuffPost feed', () => {
    it('should not crash when parsing the feed', async () => {
      const result = rssParser.parseFeed(huffpost.feed);
      expect(result).not.toBe(undefined);
    });
  });

  describe('when item has no updated element', () => {
    it('should return published element', async () => {
      const result = rssParser.parseFeed(atomv1NoUpdated.feed);
      expect(
        result.items[0].published instanceof Date
          ? result.items[0].published.toISOString()
          : result.items[0].published,
      ).toContain('2016-01-01');
    });
  });
});
