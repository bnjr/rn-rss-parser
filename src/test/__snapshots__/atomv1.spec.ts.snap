// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when parse ATOM valid document should return feed items 1`] = `
{
  "authors": [],
  "categories": [
    {
      "label": "Information",
      "term": "Information",
      "url": null,
    },
    {
      "label": "Information",
      "term": "Information",
      "url": null,
    },
  ],
  "copyright": null,
  "description": "A sample ATOM feed",
  "generator": {
    "name": "Jekyll",
    "url": "http://jekyllrb.com",
    "version": "3.1.2",
  },
  "image": {
    "title": null,
    "url": "https://b.thumbs.redditmedia.com/ntr1FkBiO3nk4t4Vgy5GXoPQ_j2hirENH9iT8rXNf8M.png",
  },
  "items": [
    {
      "authors": [],
      "categories": [
        {
          "label": "Information",
          "term": "Information",
          "url": null,
        },
      ],
      "content": "<p>The chocolate chip cookie was invented by Ruth Graves Wakefield. She owned the Toll House Inn, in Whitman, Massachusetts, a very popular restaurant that featured home cooking in the 1930s. Her cookbook, Toll House Tried and True Recipes, was first published in 1936 by M. Barrows &amp; Company, New York. The 1938 edition of the cookbook was the first to include the recipe "Toll House Chocolate Crunch Cookie" which rapidly became a favorite cookie in American homes.</p>

<p>Source / Read more <a href="https://en.wikipedia.org/wiki/Chocolate_chip_cookie">Wikipedia</a></p>
    ",
      "description": "The chocolate chip cookie was invented by Ruth Graves Wakefield.",
      "id": "http://bakery-store.example.com/information/2016/01/02/where-did-the-cookie-come-from",
      "image": null,
      "media": [
        {
          "image": null,
          "length": 1234,
          "mimeType": "audio/mpeg",
          "title": null,
          "type": "audio",
          "url": "https://www.example.com/audio.mp3",
        },
      ],
      "published": "2016-01-01T11:00:00.000Z",
      "title": "Where Did The Cookie Come From",
      "updated": "2016-01-01T11:00:00.000Z",
      "url": "http://bakery-store.example.com/information/2016/01/02/where-did-the-cookie-come-from.html",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "Information",
          "term": "Information",
          "url": null,
        },
      ],
      "content": "<p>Sourdough bread is made by the fermentation of dough using naturally-occurring lactobacilli and yeast. Sourdough bread has a mildly sour taste not present in most breads made with baker's yeast and better inherent keeping qualities than other breads, due to the lactic acid produced by the lactobacilli.</p>

<p>Source / Read more <a href="https://en.wikipedia.org/wiki/Sourdough">Wikipedia</a></p>
    ",
      "description": "Sourdough bread is made by the fermentation of dough using naturally-occurring lactobacilli and yeast. Sourdough bread has a mildly sour taste not present in most breads made with baker's yeast and better inherent keeping qualities than other breads, due to the lactic acid produced by the lactobacilli.",
      "id": "http://bakery-store.example.com/information/2016/01/01/what-is-sour-dough",
      "image": null,
      "media": [],
      "published": "2015-12-31T11:00:00.000Z",
      "title": "What Is Sour Dough",
      "updated": "2015-12-31T11:00:00.000Z",
      "url": "http://bakery-store.example.com/information/2016/01/01/what-is-sour-dough.html",
    },
  ],
  "language": null,
  "meta": {
    "type": "atom",
    "version": "1.0",
  },
  "published": "2016-01-01T11:00:00.000Z",
  "self": null,
  "title": "ATOM title",
  "updated": "2016-05-13T04:22:08.000Z",
  "url": "http://bakery-store.example.com/",
}
`;

exports[`when parse ATOM with itunes elements should return itunes information for channel and item elements 1`] = `
{
  "authors": [],
  "categories": [
    {
      "label": "Information",
      "term": "Information",
      "url": null,
    },
    {
      "label": "Information",
      "term": "Information",
      "url": null,
    },
  ],
  "copyright": null,
  "description": "A sample ATOM feed",
  "generator": {
    "name": "Jekyll",
    "url": "http://jekyllrb.com",
    "version": "3.1.2",
  },
  "image": null,
  "items": [
    {
      "authors": [],
      "categories": [
        {
          "label": "Information",
          "term": "Information",
          "url": null,
        },
      ],
      "content": "<p>The chocolate chip cookie was invented by Ruth Graves Wakefield. She owned the Toll House Inn, in Whitman, Massachusetts, a very popular restaurant that featured home cooking in the 1930s. Her cookbook, Toll House Tried and True Recipes, was first published in 1936 by M. Barrows &amp; Company, New York. The 1938 edition of the cookbook was the first to include the recipe "Toll House Chocolate Crunch Cookie" which rapidly became a favorite cookie in American homes.</p>

<p>Source / Read more <a href="https://en.wikipedia.org/wiki/Chocolate_chip_cookie">Wikipedia</a></p>
      ",
      "description": "The chocolate chip cookie was invented by Ruth Graves Wakefield.",
      "id": "http://bakery-store.example.com/information/2016/01/02/where-did-the-cookie-come-from",
      "image": null,
      "media": [
        {
          "image": null,
          "length": 1234,
          "mimeType": "audio/mpeg",
          "title": null,
          "type": "audio",
          "url": "https://www.example.com/audio.mp3",
        },
      ],
      "published": "2016-01-01T11:00:00.000Z",
      "title": "Where Did The Cookie Come From",
      "updated": "2016-01-01T11:00:00.000Z",
      "url": "http://bakery-store.example.com/information/2016/01/02/where-did-the-cookie-come-from.html",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "Information",
          "term": "Information",
          "url": null,
        },
      ],
      "content": "<p>Sourdough bread is made by the fermentation of dough using naturally-occurring lactobacilli and yeast. Sourdough bread has a mildly sour taste not present in most breads made with baker's yeast and better inherent keeping qualities than other breads, due to the lactic acid produced by the lactobacilli.</p>

<p>Source / Read more <a href="https://en.wikipedia.org/wiki/Sourdough">Wikipedia</a></p>
      ",
      "description": "Sourdough bread is made by the fermentation of dough using naturally-occurring lactobacilli and yeast. Sourdough bread has a mildly sour taste not present in most breads made with baker's yeast and better inherent keeping qualities than other breads, due to the lactic acid produced by the lactobacilli.",
      "id": "http://bakery-store.example.com/information/2016/01/01/what-is-sour-dough",
      "image": null,
      "media": [],
      "published": "2015-12-31T11:00:00.000Z",
      "title": "What Is Sour Dough",
      "updated": "2015-12-31T11:00:00.000Z",
      "url": "http://bakery-store.example.com/information/2016/01/01/what-is-sour-dough.html",
    },
  ],
  "language": null,
  "meta": {
    "type": "atom",
    "version": "1.0",
  },
  "published": "2016-01-01T11:00:00.000Z",
  "self": null,
  "title": "ATOM title",
  "updated": "2016-05-13T04:22:08.000Z",
  "url": "http://bakery-store.example.com/",
}
`;
